code/game/dna/genes/disabilities.dm

```diff
@@ -249,14 +249,14 @@
 	block = GLOB.wingdingsblock
 
 /datum/dna/gene/disability/wingdings/OnSay(mob/M, message)
-	var/list/chars = string2charlist(message)
 	var/garbled_message = ""
-	for(var/C in chars)
-		if(C in GLOB.alphabet_uppercase)
+	var/messagelength = length(message)
+	for(var/i = 1; i <= messagelength; i++)
```

Comment:
This style of loop is slower
```dm
/proc/test1()
    for(var/i = 0; i < 500000; i++)
        var/e = i

/proc/test2()
    for(var/i in 1 to 500000)
        var/e = i

/proc/main()
    BENCHT("1", 1, test1())
    BENCHT("2", 1, test2())
```
Outputs
```
1: 57.857 ms
2: 26.734 ms
```

Please use something along the lines of
`for(var/i in 1 to length(message))`

Reply:
done